src/mimswap/periphery/Router.sol

```diff
@@ -481,23 +512,29 @@ contract Router {
         uint256 baseInAmount,
         uint256 quoteInAmount
     ) internal view returns (uint256 baseAdjustedInAmount, uint256 quoteAdjustedInAmount) {
-        if (IERC20(lp).totalSupply() == 0) {
+        (uint256 baseReserve, uint256 quoteReserve) = IMagicLP(lp).getReserves();
+
+        uint256 baseBalance = IMagicLP(lp)._BASE_TOKEN_().balanceOf(address(lp)) + baseInAmount;
+        uint256 quoteBalance = IMagicLP(lp)._QUOTE_TOKEN_().balanceOf(address(lp)) + quoteInAmount;
+
+        baseInAmount = baseBalance - baseReserve;
+        quoteInAmount = quoteBalance - quoteReserve;
+
+        if (quoteReserve == 0 && baseReserve == 0) {
```

Comment:
IMO I think this case should still match up with the totalSupply == 0 case in the MagicLp buyShares function to avoid any weird behavior

Reply:
fixed in dcf6037685e4242c2b0a5131f889ba77c07cd6ab

make sense now?